packages/Webkul/API/Http/Controllers/Shop/AddressController.php

```diff
@@ -104,26 +106,28 @@ public function store()
      *
      * @return \Illuminate\Http\Response
      */
-    public function update()
+    public function update($id)
     {
         $customer = auth($this->guard)->user();
 
         request()->merge(['address1' => implode(PHP_EOL, array_filter(request()->input('address1')))]);
 
         $this->validate(request(), [
             'address1' => 'string|required',
-            'country'  => 'string|required',
-            'state'    => 'string|required',
-            'city'     => 'string|required',
+            'company' => 'string|nullable',
+            'vat_id' => 'string|nullable',
+            'country' => 'string|required',
+            'state' => 'string|nullable',
+            'city' => 'string|required',
             'postcode' => 'required',
-            'phone'    => 'required',
+            'phone' => 'required',
         ]);
 
-        $this->customerAddressRepository->update(request()->all(), request()->input('id'));
+        $this->customerAddressRepository->update(request()->all(), $id);
 
         return response()->json([
             'message' => 'Your address has been updated successfully.',
-            'data'    => new CustomerAddressResource($this->customerAddressRepository->find(request()->input('id'))),
+            'data'    => new CustomerAddressResource($this->customerAddressRepository->find($id)),
```

Comment:
And pass that variable here. No need to run find query again.

Reply:
If I just pass it like that:
'data'    => new CustomerAddressResource($id),

I get this error:
 "message": "Trying to get property 'id' of non-object",
 From DelegatesToResource